### ISSUE TITLE
exported GCT does not contain all metadata tracks

### ISSUE THREAD

**Issue by mhk658** [cmap/cmapR#41]
I'm using write_gct() to export a GCT object with 8 tracks of column metadata.  When I open the exported file, only one metadata track is displayed (in additional to column names), even though the dimensions in the file denote 8.

Inspecting the GCT object itself shows that all 8 tracks are there in cdesc.  Any idea why this might be happening?  Thanks!

**Comment by tnat1031** [comment]
Hi @mhk658, could you please include some code to indicate how your GCT object is being constructed? That will help me debug.

Thanks a lot,
Ted

**Comment by mhk658** [comment]
Hi Ted -- I'm using the following code, from a data.frame `filtered.data` and a `annotation.matrix` tab-delimited table with the first column named "id" containing sample names :

```
mat <- data.matrix(filtered.data[,(metadata.cols+1):(ncol(filtered.data))]) #data only
cdesc <- read.delim(annotation.matrix, header = TRUE, sep = '\t', colClasses = 'character')
rdesc <- data.frame(filtered.data[,1:metadata.cols]) #metadata only

colnames(mat) <- cdesc[,1]
rownames(mat) <- filtered.data[,1]

for(col in c(1:ncol(rdesc))){
  rdesc[,col] <- as.character(rdesc[,col])
}

output.GCT <- new("GCT", mat=mat, rdesc=rdesc, cdesc=cdesc)
write_gct(output.GCT, paste(output.file.name, '.gct', sep = ''), appenddim = FALSE)
```

I've coerced everything in `cdesc` and `rdesc` to characters in hopes that it would help, but it didn't.  If I `print(output.GCT)`, it appears to look like it says it should at https://github.com/cmap/cmapR/blob/master/cmapR_tutorial.ipynb.  The issue only appears in the exported GCT, and only in the column metadata (row metadata is fine) Please let me know if any other info would be helpful!

Thanks,
Harry

**Comment by tnat1031** [comment]
Thanks Harry. Are there any `NA` values in `cdesc`? Would it be possible to share the underlying table?

**Comment by mhk658** [comment]
There were some NA values, but I've changed them and the issue still persists.  I've attached the table used to create `cdesc`.

[annotation_matrix.txt](https://github.com/cmap/cmapR/files/4237992/annotation_matrix.txt)

**Comment by tnat1031** [comment]
Hi @mhk658, thanks for sending this file. I was able to reproduce the issue. Turns out there was a small bug in the `write_gct` code causing the problem. I pushed a fix with this commit: 18293e6
Could you please try pulling the updated code and seeing that it works for you?

Thanks a lot,
Ted

**Comment by mhk658** [comment]
Hi Ted -- the issue no longer occurs with the updated code.  Thanks for the help!
